src/Storages/System/StorageSystemClusters.cpp

```diff
@@ -77,6 +85,12 @@ void StorageSystemClusters::writeCluster(MutableColumns & res_columns, const Nam
             res_columns[i++]->insert(pool_status[replica_index].error_count);
             res_columns[i++]->insert(pool_status[replica_index].slowdown_count);
             res_columns[i++]->insert(pool_status[replica_index].estimated_recovery_time.count());
+            res_columns[i++]->insert(address.database_shard_name);
+            res_columns[i++]->insert(address.database_replica_name);
+            if (is_active.empty())
+                res_columns[i++]->insertDefault();
+            else
+                res_columns[i++]->insert(is_active[replica_idx++]);
```

Comment:
This part is difficult to understand: there is replica_index, and also replica_idx, and for some reason, it is incremented for every shard and replica, but only if there is `is_active`...